### ISSUE TITLE
Export ingest pipelines offline without contact

### ISSUE THREAD

**Issue by 111andre111** [elastic/beats#22619]
**Describe the enhancement:**

As of today there is an offer to export ingest pipeline when output to logstash for instance you once need to have access to elasticsearch itself with the beats in order to install the needed ingest pipelines. 
https://www.elastic.co/guide/en/beats/filebeat/current/load-ingest-pipelines.html
It would be really neat if you could export these using for instance a `filebeat export pipelines` command to json files, that you can import later on using the elasticsearch APIs.

**Describe a specific use case for the enhancement or feature:**
Today you are able to import all other other kinds of setup type that the beats use like
dashboards
ilm-policies
index-pattern
index-templates

So the pipelines are the only type that you cannot export and then load manually.
In certain company setups this might be a huge increase  in usage if these is no need to connect directly to elasticsearch with the beats.

**Comment by strawgate** [comment]
Yes please!

Right now we need to have filebeat import them and then export them which isn't particularly fun!

**Comment by nimarezainia** [comment]
@strawgate and @111andre111 would you be able to out line the workflow you currently would perform? I want to make sure we are addressing your workflow with any enhancement we pursue.

**Comment by strawgate** [comment]
Yeah,

We do not let agents talk directly to Elasticsearch today. We always use Logstash and thus load the index templates and pipelines via a node bootstrap process.

As a result we have to get the pipelines from the beat somehow and then import the index templates, component templates and pipelines into Elasticsearch.

**Comment by kvch** [comment]
> As a result we have to get the pipelines from the beat somehow and then import the index templates, component templates and pipelines into Elasticsearch.

How do you export the pipeline now? How do you import the templates to ES? Are you using the [PUT Pipeline API](https://www.elastic.co/guide/en/elasticsearch/reference/current/put-pipeline-api.html)? Do you need the pipelines to be exported in JSON format?

What I have in mind is introducing a new subcommand for `export` that would accept a pipeline ID and then print the exported pipeline in JSON format to the stdout, similarly to other `export` subcommands.

Example:

```shell
$ ./filebeat export pipeline my-pipeline-id
{
    "description": "Ingest pipeline for my data",
    "processors": [],
}
```

**Comment by strawgate** [comment]
Yeah we download the beat, run it on the command line with a relatively empty config -- something like this

`& ".\$Beat.exe" setup --pipelines --modules ios,asa -E "output.elasticsearch.hosts=['http://$ESHost`:$ESPort']" -c ".\filebeat.yml"`

Once they are imported we then export them using a GET to ES and then save each one to a file.

Later on our bootstrapping process grabs the saved pipeline JSON files and then we use the PUT Pipeline API to import them.

**Comment by 111andre111** [comment]
Yes @kvch that was exactly what I had in mind.

@nimarezainia the use case has been described in one of the last comments similar to mine.

**Comment by swtrux** [comment]
Is this export option for pipelines in work?  coming soon?  This would be very useful for us as well as sometimes we need to adjust the baseline ingest pipeline for use in our environment before loading into Elasticsearch.

**Comment by nimarezainia** [comment]
@strawgate @swtrux would it be possible to have you migrate to Elastic Agent where handling of pipelines is managed for you?  I would like to know if you have a specific use case that may inhibit the migration to Agent and perhaps work with you to support your deployment model. Please let us know.

**Comment by YiuTerran** [comment]
any update?
